sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.applicationChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,307 @@
+  });
+
+  beforeEach(async () => {
+    MockDate.reset();
+  });
+
+  it("Should create a change request when the application is completed.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const completedApplication = await saveFakeApplication(
+      db.dataSource,
+      { student, programYear },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    const endpoint = `/students/application/${completedApplication.id}/change-request`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    await mockUserLoginInfo(appModule, student);
+    const now = new Date();
+    MockDate.set(now);
+
+    // Act/Assert
+    let createApplicationId: number;
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(defaultPayload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        createApplicationId = response.body.id;
+        expect(createApplicationId).toBeGreaterThan(0);
+      });
+    const createdApplication = await db.application.findOne({
+      select: {
+        id: true,
+        submittedDate: true,
+        applicationNumber: true,
+        relationshipStatus: true,
+        studentNumber: true,
+        programYear: { id: true },
+        offeringIntensity: true,
+        location: { id: true },
+        parentApplication: { id: true },
+        precedingApplication: { id: true },
+        student: { id: true },
+        applicationStatus: true,
+        applicationStatusUpdatedOn: true,
+        applicationEditStatus: true,
+        applicationEditStatusUpdatedOn: true,
+        applicationEditStatusUpdatedBy: { id: true },
+        currentAssessment: {
+          id: true,
+          studentAssessmentStatus: true,
+          studentAssessmentStatusUpdatedOn: true,
+          offering: { id: true },
+          studentAppeal: { id: true },
+          creator: { id: true },
+          createdAt: true,
+        },
+      },
+      relations: {
+        programYear: true,
+        location: true,
+        parentApplication: true,
+        precedingApplication: true,
+        student: true,
+        applicationEditStatusUpdatedBy: true,
+        currentAssessment: {
+          offering: true,
+          studentAppeal: true,
+          creator: true,
+        },
+      },
+      where: {
+        id: createApplicationId,
+      },
+      loadEagerRelations: false,
+    });
+    // Expected user object.
+    const expectedUserObject = expect.objectContaining({
+      id: student.user.id,
+    });
+    expect(createdApplication).toEqual({
+      id: createApplicationId,
+      submittedDate: now,
+      applicationNumber: completedApplication.applicationNumber,
+      relationshipStatus: defaultPayload.data.relationshipStatus,
+      studentNumber: defaultPayload.data.studentNumber,
+      programYear: expect.objectContaining({
```

Comment:
Do we have to use the `expect.objectContaining` here?

Reply:
Removed `expect.objectContaining` on both E2E files. They were added by mistake due to another error in the SQL queries that were returning more data than they should. When the data is right, there is no reason to use `expect.objectContaining`.
Nice catch!